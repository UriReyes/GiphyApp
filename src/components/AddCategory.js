import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmint = (e) => {
    e.preventDefault();
    if (inputValue !== undefined) {
      if (inputValue.trim().length > 2) {
        setCategories((cats) => [inputValue, ...cats]);
        setInputValue("");
      } else {
        alert("No has escrito nada");
      }
    }
  };

  return (
    <>
      <form className="formulario" onSubmit={handleSubmint}>
        <div className="d-flex-row">
          <input
            type="text"
            placeholder="Buscar"
            value={inputValue}
            onChange={handleInputChange}
          />
          <button
            className="d-flex-row align-items-center btn btn-search"
            onClick={handleSubmint}
          >
            <p>ðŸ”Ž</p>
          </button>
        </div>
      </form>
      <br />
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
