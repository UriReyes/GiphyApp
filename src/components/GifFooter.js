import React from "react";

export const GifFooter = () => {
  return (
    <div className="footer">
      <div className="d-flex-row align-items-center justify-content-center">
        <h4 className="footer-app">Designed by UriCode &copy;</h4>
        <img
          className="img-footer"
          src={process.env.PUBLIC_URL + "helmet.svg"}
          alt="img_footer"
        />
      </div>
      <p className="text-center">
        <small>
          Este es un proyecto construido con el API de Giphy, todos los Gifs que
          aqu√≠ aparecen son pertenecientes a Giphy.
        </small>
      </p>
    </div>
  );
};
