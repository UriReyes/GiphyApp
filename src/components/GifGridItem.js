import React from "react";
import PropTypes from "prop-types";

import { useLoadingGif } from "../hooks/useLoadingGif";

export const GifGridItem = ({ title, url }) => {
  const gifData = useLoadingGif(url);

  const {
    event: imageLoaded,
    data: { gif, loaded },
  } = gifData;

  const showInformationHandler = (e) => {
    const cardInfo = e.currentTarget.firstChild;
    const cardInfoClasses = cardInfo.classList;

    if (cardInfoClasses.contains("d-none")) {
      cardInfoClasses.remove("d-none");
    }
  };

  const hideInformationHandler = (e) => {
    const cardInfo = e.currentTarget.firstChild;
    const cardInfoClasses = cardInfo.classList;

    if (!cardInfoClasses.contains("d-none")) {
      cardInfoClasses.add("d-none");
    }
  };

  return (
    <div
      className="d-flex justify-content-center flex-wrap card animate__animated animate__fadeIn n-father"
      onMouseOver={showInformationHandler}
      onMouseLeave={hideInformationHandler}
    >
      <div className="n-child d-none d-flex-row align-items-bottom justify-content-between">
        <p className="text-capitalize text-title card-info">{title}</p>
        <a
          href={url}
          className="btn card-info"
          target="_blank"
          rel="noreferrer"
        >
          ðŸ”—
        </a>
      </div>
      <img
        src={url}
        alt={title}
        onLoad={imageLoaded}
        srcSet={loaded ? gif : process.env.PUBLIC_URL + "loading.gif"}
      />
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
