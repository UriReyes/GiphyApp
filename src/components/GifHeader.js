import React from "react";
import PropTypes from "prop-types";

import { AddCategory } from "./AddCategory";

export const GifHeader = ({ setCategories }) => {
  return (
    <div className="nav d-flex align-items-center">
      <img
        src="https://media.giphy.com/media/3o6gbbuLW76jkt8vIc/giphy.gif"
        alt="img-header"
      />
      <div className="d-flex justify-content-center w-100">
        <AddCategory setCategories={setCategories} />
      </div>
    </div>
  );
};

GifHeader.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
