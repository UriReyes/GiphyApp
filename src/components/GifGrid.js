import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div className="container text-center">
      <h3 className="text-capitalize">{category}</h3>
      <hr />
      <br />
      {loading && (
        <div className="lds-roller">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      )}
      <div className="d-flex flex-wrap justify-content-between">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </div>
  );
};
