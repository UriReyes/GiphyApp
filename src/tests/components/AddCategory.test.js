import React from "react";

import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Probando <AddCategory/>", () => {
  const key = "";
  const setCategories = jest.fn();
  let wrapper = shallow(
    <AddCategory key={key} setCategories={setCategories} />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory key={key} setCategories={setCategories} />);
  });

  test("Debe de renderizar el <AddCategory /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";
    const event = { target: { value: value } };
    input.simulate("change", event);
    //console.log(input.prop("value"));
    // expect(input.prop("value")).toBe(value);
  });

  test("No debe de postear informaciÃ³n onSubmit", () => {
    const event = { preventDefault: () => {} };
    const jsdomAlert = window.alert; // remember the jsdom alert
    window.alert = () => {}; // provide an empty implementation for window.alert
    wrapper.find("form").simulate("submit", event);

    expect(setCategories).not.toHaveBeenCalled();
    window.alert = jsdomAlert; // restore the jsdom alert
  });

  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {
    // Simulando el cambio en el input
    const value = "Naruto";
    const input = wrapper.find("input");
    const eventInput = { target: { value } };
    input.simulate("change", eventInput);

    // Simulando el submit del formulario
    const eventFormulario = { preventDefault: () => {} };
    wrapper.find("form").simulate("submit", eventFormulario);

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop("value")).toBe("");
  });
});
