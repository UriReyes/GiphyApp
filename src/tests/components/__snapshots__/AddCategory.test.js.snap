// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <AddCategory/> Debe de renderizar el <AddCategory /> correctamente 1`] = `
<Fragment>
  <form
    className="formulario"
    onSubmit={[Function]}
  >
    <div
      className="d-flex-row"
    >
      <input
        onChange={[Function]}
        placeholder="Buscar"
        type="text"
        value=""
      />
      <button
        className="d-flex-row align-items-center btn btn-search"
        onClick={[Function]}
      >
        <p>
          ðŸ”Ž
        </p>
      </button>
    </div>
  </form>
  <br />
</Fragment>
`;
